lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -600,19 +605,20 @@ def write(chunk)
           entry_common_labels = \
             group_common_labels.merge(extracted_common_labels)
 
-          if entry_resource.type == CONTAINER_CONSTANTS[:resource_type]
+          is_json = false
+          if @detect_json
             # Save the timestamp if available, then clear it out to allow for
             # determining whether we should parse the log or message field.
-            timestamp = record.key?('time') ? record['time'] : nil
-            record.delete('time')
+            timestamp = record.delete('time')
```

Comment:
Just confirming - this a more concise way of doing the 2 lines you replaced?

Reply:
Exactly: [Hash#delete](https://docs.ruby-lang.org/en/2.0.0/Hash.html#method-i-delete).